bin/check_multi-line-string-array_contents

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# This script checks that multi-line-string-array only contains valid keys.
+
+failed=0
+
+check_keys() {
+    json_file=$1
+    echo "Checking 'multi-line-string-array' contents in $json_file"
+
+    present_keys=$(jq '[ .cases[] | recurse(.cases[]?) | ."multi-line-string-array" | select(. != null)[] ] | unique ' $json_file)
+
+    allowed_keys=$(jq '[ .cases[] | recurse(.cases[]?) | .input | keys[] ] | unique + ["expected"]' $json_file)
```

Comment:
Here it *does* make sense to `recurse(.cases[]?)` if we assume that an outer `multi-line-string-array` key applies to nested cases in a `labeledTestGroup`.

I tried to run this on a canonical-data.json picked at random (high-scores), and it resulted in an error:

```
$ jq '[ .cases[] | recurse(.cases[]?) | .input | keys[] ] | unique + ["expected"]' exercises/high-scores/canonical-data.json
jq: error (at exercises/high-scores/canonical-data.json:83): null (null) has no keys
```

The problem here is that `labeledTestGroup` does not have direct `.input` fields, so this produces `null` values that, naturally, don't have any `keys`. The nested keys are still addressed by `recurse(.cases[]?)`.

One fix is to filter out those `null`s.

---

> What if we had an input `expected`?

> We shouldn't and I suggest explicitly denying that from the input list.

But we don't currently deny that, so *what if we had?*

Then `"expected"` might occur in the recursive search, potentially producing

```
[ "input1", "expected", "input3", "expected" ]
```

because `"expected"` is added to the allowed set *after* `unique`.

```suggestion
    allowed_keys=$(jq '[ .cases[] | recurse(.cases[]?) | .input | select(. != null) | keys[] ] + ["expected"] | unique' $json_file)
```